File: frontend/be_elite/lib/models/Program/post_invite_dto.dart
class PostInviteDto {
	String? programName;
	String? athleteUsername;
  String? coachId;

	PostInviteDto({this.programName, this.athleteUsername, this.coachId});

	factory PostInviteDto.fromJson(Map<String, dynamic> json) => PostInviteDto(
				programName: json['programName'] as String?,
				athleteUsername: json['athleteUsername'] as String?,
        coachId: json['coachId'] as String?,
			);

	Map<String, dynamic> toJson() => {
				'programName': programName,
				'athleteUsername': athleteUsername,
        'coachId': coachId,
			};
}

File: backend/BeElite/src/main/java/com/salesianos/triana/BeElite/dto/Program/PostInviteDto.java
package com.salesianos.triana.BeElite.dto.Program;

import com.salesianos.triana.BeElite.model.Invite;
import lombok.Builder;

import java.util.List;
import java.util.UUID;

@Builder
public record PostInviteDto(
        String athleteUsername,
        String programName,
        UUID coachId
) {

    public static PostInviteDto of(Invite i) {
        return PostInviteDto.builder()
                .athleteUsername(i.getAthlete().getUsername())
                .programName(i.getProgram().getProgramName())
                .coachId(i.getProgram().getCoach().getId())
                .build();
    }
}
